From: =?utf-8?q?IOhannes_m_zm=C3=B6lnig?= <umlaeute@debian.org>
Date: Tue, 29 Jan 2019 11:31:36 +0100
Subject: fix menu-entries in case puredata-doc is not available

if the puredata-doc package is not installed, some of the menu-entries (namely
the manual section) will not work correctly.
This patch checks whether the files are there, and if not, a dialog is raised
asking the user to install puredata-doc

Origin: Debian
Forwarded: not-needed
---
 tcl/pd_menucommands.tcl | 14 ++++++++++++++
 1 file changed, 14 insertions(+)

diff --git a/tcl/pd_menucommands.tcl b/tcl/pd_menucommands.tcl
index 954986f..eb76d85 100644
--- a/tcl/pd_menucommands.tcl
+++ b/tcl/pd_menucommands.tcl
@@ -232,10 +232,23 @@ proc ::pd_menucommands::set_filenewdir {mytoplevel} {
     }
 }
 
+proc ::pd_menucommands::check_puredatadoc {file} {
+  if { [string compare "absolute" [file pathtype $file]] != 0 } {return true}
+  if { [file exists $file] } {
+	return true
+  } {
+    tk_messageBox -type ok -message "Unable to find '$file'.\nMaybe you need to install the 'puredata-doc' package"
+    return false
+  }
+}
+
+
 # parse the textfile for the About Pd page
 proc ::pd_menucommands::menu_aboutpd {} {
     set versionstring "Pd $::PD_MAJOR_VERSION.$::PD_MINOR_VERSION.$::PD_BUGFIX_VERSION$::PD_TEST_VERSION"
     set filename "$::sys_libdir/doc/1.manual/1.introduction.txt"
+    set filename "/usr/share/puredata-gui/1.introduction.txt"
+    if {![check_puredatadoc $filename]} return
     if {![file exists $filename]} {
         ::pdwindow::error [_ "ignoring '%s': doesn't exist" $filename]
         ::pdwindow::error "\n"
@@ -295,6 +308,7 @@ proc ::pd_menucommands::menu_doc_open {dir basename} {
 
 # open HTML docs from the menu using the OS-default HTML viewer
 proc ::pd_menucommands::menu_openfile {filename} {
+    if {![check_puredatadoc $filename]} return
     if {$::tcl_platform(os) eq "Darwin"} {
         exec sh -c [format "open '%s'" $filename]
     } elseif {$::tcl_platform(platform) eq "windows"} {
